edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/KubernetesEnvironmentOperatorTest.cs

```diff
@@ -0,0 +1,151 @@
+
+        readonly KubernetesRuntimeInfoProvider runtimeInfoProvider;
+
+        static readonly TimeSpan DefaultTimeout = TimeSpan.FromMinutes(1);
+
+        public KubernetesEnvironmentOperatorTest(KubernetesClusterFixture fixture)
+        {
+            string deviceNamespace = $"device-{Guid.NewGuid()}";
+            this.client = new KubernetesClient(deviceNamespace, fixture.Client);
+
+            this.runtimeInfoProvider = new KubernetesRuntimeInfoProvider(deviceNamespace, fixture.Client, new FakeModuleManager());
+            this.environmentOperator = new KubernetesEnvironmentOperator(deviceNamespace, this.runtimeInfoProvider, fixture.Client);
+        }
+
+        public async Task InitializeAsync()
+        {
+            await this.client.AddNamespace();
+
+            this.environmentOperator.Start();
+        }
+
+        public Task DisposeAsync()
+        {
+            this.environmentOperator?.Dispose();
+            return Task.CompletedTask;
+        }
+
+        [Fact]
+        public async Task CollectsModuleRuntimeInfoWhenModuleDeploymentAdded()
+        {
+            var tokenSource = new CancellationTokenSource(DefaultTimeout);
+
+            await this.AddEdgeModule("Module-A");
+            await this.client.WaitUntilAnyPods("status.phase=Running", tokenSource.Token);
+
+            ModuleRuntimeInfo moduleInfo = await this.GetModule("Module-A");
+
+            moduleInfo.Should().NotBeNull();
+            moduleInfo.ModuleStatus.Should().Be(ModuleStatus.Running);
+            moduleInfo.Name.Should().Be("Module-A");
+            moduleInfo.Description.Should().StartWith("Started at");
+            moduleInfo.Type.Should().Be("docker");
+            moduleInfo.ExitCode.Should().Be(0);
+            moduleInfo.StartTime.ShouldOption().NotBeNone();
+            moduleInfo.ExitTime.ShouldOption().BeNone();
+        }
+
+        [Fact]
+        public async Task DoNotCollectModuleRuntimeInfoForUnknownModules()
+        {
+            var tokenSource = new CancellationTokenSource(DefaultTimeout);
+
+            var labels = new Dictionary<string, string> { ["a"] = "b" };
+            await this.client.AddModuleServiceAccount("foreign-pod", labels, null);
+            await this.client.AddModuleDeployment("foreign-pod", labels, null);
+            await this.client.WaitUntilAnyPods("status.phase=Running", tokenSource.Token);
+
+            ModuleRuntimeInfo moduleInfo = await this.GetModule("foreign-pod");
+
+            moduleInfo.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task DeletesModuleRuntimeInfoForDeletedModules()
+        {
+            var tokenSource = new CancellationTokenSource(DefaultTimeout * 3);
+
+            await this.AddEdgeModule("Module-A");
+            await this.client.WaitUntilAnyPods("status.phase=Running", tokenSource.Token);
+
+            IEnumerable<ModuleRuntimeInfo> modules = await this.runtimeInfoProvider.GetModules(CancellationToken.None);
```

Comment:
>CancellationToken.None [](start = 95, length = 22)

I was thinking this should be a timeout as well, but I'm not convinced it's needed.

Reply:
I don't think it is required here: we just take whatever we have in memory; no async operations